bokeh/charts/chart.py

```diff
@@ -82,14 +82,14 @@ class Chart(Plot):
     def __init__(self):
 
         # Initializes then gets default properties
-        super(Chart, self).__init__(id=self._options.id or make_id())
+        # NOTE: we use option._id here since ChartOptions converts id arg to _id
+        super(Chart, self).__init__(id=self._options._id or make_id())
 
         # create new chart options to get default properties with values
         default_props = ChartOptions().properties_with_values()
 
         # get the properties and values from the module-wide defaults object
         option_props = ChartOptions.properties_with_values(defaults)
```

Comment:
I know it isn't quite related to your patch but I'm wondering about this code.

1. `defaults` is a default-constructed ChartOptions. So default_props and option_props should always be identical ? I guess the idea is that someone can change the defaults object?
2. Why `ChartOptions.properties_with_values(defaults)` instead of `defaults.properties_with_values()`?
3. if `bokeh.charts.chart_options.default_options` has been modified, it looks like the values in there override options that are explicitly passed in to this `Chart.__init__` - is that intentional? Shouldn't my constructor kwargs override the `default_options`?

There's maybe a better way to write this using the new `include_defaults` param to `properties_with_values()`, that is instead of comparing each value in `option_props` to the defaults, you could do `option_props = defaults.properties_with_values(include_defaults=False)`, to get only those that have been modified. In that case I think there's no need to create `default_props` at all.